### PR TITLE
W2 - hieucap - create recyclerview person screen

### PR DESCRIPTION
**Pull request by at-hieucap**
#### Related issues (Optional)
Task #21 

#### Description (Required)
- [x] Create RecyclerView

### CODE REVIEW

File: app/src/at-hieucap/java/vn/asiantech/internship/calculator/CalculatorActivity.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        mBtnMul.setOnClickListener(this);
+        mBtnDivision.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()) {
+            case R.id.btnSum:
+                if (checkAllTermIsEmpty(mEdtFirstTerm, mEdtSecondTerm)) {
+                    Toast.makeText(CalculatorActivity.this, "First term and second term no empty !", Toast.LENGTH_LONG).show();
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/calculator/CalculatorActivity.java]: WARNING: [AvoidDuplicateLiterals] The String literal "First term and second term no empty !" appears 4 times in this file; the first occurrence is on line 44
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/at-hieucap/java/vn/asiantech/internship/calculator/CalculatorActivity.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                } else if (checkFirstTermIsEmpty(mEdtFirstTerm, mEdtSecondTerm)) {
+                    Toast.makeText(CalculatorActivity.this, "First term no empty !", Toast.LENGTH_LONG).show();
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/calculator/CalculatorActivity.java]: WARNING: [AvoidDuplicateLiterals] The String literal "First term no empty !" appears 4 times in this file; the first occurrence is on line 46
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/at-hieucap/java/vn/asiantech/internship/calculator/CalculatorActivity.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                } else if (checkSecondTermIsEmpty(mEdtFirstTerm, mEdtSecondTerm)) {
+                    Toast.makeText(CalculatorActivity.this, "Second term no empty !", Toast.LENGTH_LONG).show();
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/calculator/CalculatorActivity.java]: WARNING: [AvoidDuplicateLiterals] The String literal "Second term no empty !" appears 4 times in this file; the first occurrence is on line 48
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

**Comment by at-hieucap** [comment]
@at-huongnguyen  Please reivew for me
Thank you so much.

**Comment by at-hieucap** [comment]
@at-huongnguyen em có thay đổi ở phép chia. Nhờ anh vào review giúp em với
Em cảm ơn anh.

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/Person.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
+package vn.asiantech.internship.recyclerview;
+
+public class Person {
+    private String name;
+    private boolean isLike;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/Person.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isLike has the same name as a method
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/Person.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    private boolean isUnlike;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/Person.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isUnlike has the same name as a method
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            mTvStatus.setText(person.getStatus());
+        }
+
+        @Override
+        public void onClick(View v) {
+            BigDecimal bdValue = new BigDecimal(mTvValue.getText().toString());
+            switch (v.getId()) {
+                case R.id.imgLike:
+                    mImgLike.setSelected(true);
+                    bdValue = bdValue.add(new BigDecimal(1));
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                    if (bdValue.compareTo(new BigDecimal(0)) == 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+                        mTvValue.setTextColor(ActivityCompat.getColor(mContext, R.color.colorBlack));
+                    } else if (bdValue.compareTo(new BigDecimal(0)) > 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+                        mTvValue.setTextColor(ActivityCompat.getColor(mContext, R.color.colorRed));
+                    } else {
+                        mTvValue.setTextColor(ActivityCompat.getColor(mContext, R.color.colorBlue));
+                    }
+                    mTvValue.setText(String.valueOf(bdValue));
+                    break;
+                case R.id.imgDisLike:
+                    mImgDisLike.setSelected(true);
+                    bdValue = bdValue.subtract(new BigDecimal(1));
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+                    if (bdValue.compareTo(new BigDecimal(0)) == 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+                        mTvValue.setTextColor(ActivityCompat.getColor(mContext, R.color.colorBlack));
+                    } else if (bdValue.compareTo(new BigDecimal(0)) > 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonViewActivity.java [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    private void initAdapter() {
+        mAdapter = new PersonAdapter(this, mPersonList );
+        mRecyclerViewPerson.setLayoutManager(new LinearLayoutManager(this));
+        mRecyclerViewPerson.setAdapter(mAdapter);
+    }
+
+    private void initData() {
+        mPersonList.add(new Person("Anh A", false,false,"0", "This is Sparta"));
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerview/PersonViewActivity.java]: WARNING: [AvoidDuplicateLiterals] The String literal "This is Sparta" appears 8 times in this file; the first occurrence is on line 37
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/Person.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
+package vn.asiantech.internship.recyclerView;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/Person.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/Person.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+public class Person {
+    private String name;
+    private boolean isLike;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/Person.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isLike has the same name as a method
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/Person.java [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+    private boolean isUnlike;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/Person.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isUnlike has the same name as a method
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java [hunk 16]
```diff
@@ -0,0 +1,102 @@
+package vn.asiantech.internship.recyclerView;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+            mTvStatus.setText(person.getStatus());
+        }
+
+        @Override
+        public void onClick(View v) {
+            BigDecimal bdValue = new BigDecimal(mTvValue.getText().toString());
+            switch (v.getId()) {
+                case R.id.imgLike:
+                    mImgLike.setSelected(true);
+                    bdValue = bdValue.add(new BigDecimal(1));
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+                    if (bdValue.compareTo(new BigDecimal(0)) == 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+                        mTvValue.setTextColor(ActivityCompat.getColor(mContext, R.color.colorBlack));
+                    } else if (bdValue.compareTo(new BigDecimal(0)) > 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+                        mTvValue.setTextColor(ActivityCompat.getColor(mContext, R.color.colorRed));
+                    } else {
+                        mTvValue.setTextColor(ActivityCompat.getColor(mContext, R.color.colorBlue));
+                    }
+                    mTvValue.setText(String.valueOf(bdValue));
+                    break;
+                case R.id.imgDisLike:
+                    mImgDisLike.setSelected(true);
+                    bdValue = bdValue.subtract(new BigDecimal(1));
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+                    if (bdValue.compareTo(new BigDecimal(0)) == 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+                        mTvValue.setTextColor(ActivityCompat.getColor(mContext, R.color.colorBlack));
+                    } else if (bdValue.compareTo(new BigDecimal(0)) > 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonAdapter.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonViewActivity.java [hunk 23]
```diff
@@ -0,0 +1,51 @@
+package vn.asiantech.internship.recyclerView;
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonViewActivity.java]: WARNING: [PackageCase] Package name contains upper case characters
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#PackageCase

File: app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonViewActivity.java [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    private void initAdapter() {
+        mAdapter = new PersonAdapter(this, mPersonList );
+        mRecyclerViewPerson.setLayoutManager(new LinearLayoutManager(this));
+        mRecyclerViewPerson.setAdapter(mAdapter);
+    }
+
+    private void initData() {
+        mPersonList.add(new Person("Anh A", false,false,"0", "This is Sparta"));
```
Comment (at-dev-support-01) [inline app/src/at-hieucap/java/vn/asiantech/internship/recyclerView/PersonViewActivity.java]: WARNING: [AvoidDuplicateLiterals] The String literal "This is Sparta" appears 8 times in this file; the first occurrence is on line 37
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals

**Comment by at-hieucap** [comment]
@at-huongnguyen  I finish fix comments by BOT , please review for me.
Thank you so much !

**Comment by at-huongnguyen** [comment]
@at-hieucap 
Plz resolve conflict

**Comment by at-hieucap** [comment]
@at-huongnguyen em fix conflict rồi, nhờ a vào xem lại giùm em với.
em cảm ơn anh.

**Pull request merged**